content/3_Silver/Prefix_Sums.mdx

```diff
@@ -681,6 +692,305 @@ as expected.
+#define beg(x) x.begin()
+#define en(x) x.end()
+#define all(x) beg(x), en(x)
+ 
+int N;
+ 
+int main() {
+    cin >> N;
+ 
+    vpi customers;
+ 
+    F0R(i, N) {
+        int a, b; cin >> a >> b;
+        customers.pb(mp(a, 1));
+        customers.pb(mp(b, -1));
+    }
+ 
+    sort(all(customers));
+ 
+    int count = 0;
+    int ret = 0;
+
+	//note that we don't use a prefix array in this problem because the coordinates are large.
+ 
+    trav(a, customers) {
+        count += a.s;
+        ret = max(ret, count);
+    }
+ 
+    cout << ret;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+### 2-Dimensional
+
+#### Point
+
+We can apply a similar concept to 2d matrices. Since we define $\texttt{prefix}[i][j]$ as the number of points between $(0,0)$ and $(i, j)$, each point $(i, j)$ is equal to the sum of itself and all the points between the rectangles formed by $(0, 0), (i, j-1)$ and $(0, 0),(i-1, j)$. However, since we are double counting the square $(i-1, j-1)$, we subtract it from our equation. This can be expressed by
+
+$$
+\texttt{prefix}[i][j] = \texttt{prefix}[i][j] = \texttt{prefix}[i][j] + \texttt{prefix}[i-1][j] + \texttt{prefix}[i][j-1] - \texttt{prefix}[i-1][j-1]
+$$
+
+Given the sample case in [Forest Queries](https://cses.fi/problemset/task/1652), the algorithm goes as follows:
+
+We first increment every point by 1.
+
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>1</td>
+			<td>0</td>
+			<td>0</td>
+		</tr>
+		<tr>
+			<td>1</td>
+			<td>0</td>
+			<td>1</td>
+			<td>1</td>
+		</tr>
+		<tr>
+			<td>1</td>
+			<td>1</td>
+			<td>0</td>
+			<td>0</td>
+		</tr>
+		<tr>
+			<td>1</td>
+			<td>1</td>
+			<td>1</td>
+			<td>1</td>
+		</tr>
+	</table>
+</center>
+
+Then, we apply the preprocessing step.
+
+$$
+\sum_{i=1}^{N} \sum_{j=1}^{N} \texttt{prefix}[i][j] = \texttt{prefix}[i][j] + \texttt{prefix}[i-1][j] + \texttt{prefix}[i][j-1] - \texttt{prefix}[i-1][j-1]
+$$
+
+For our example, we get 
+
+<center>
+	<table className='text-center'>
+	<tr>
+		<td>0</td>
+		<td>0</td>
+		<td>0</td>
+		<td>0</td>
+		<td>0</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>0</td>
+		<td>1</td>
+		<td>1</td>
+		<td>1</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>1</td>
+		<td>2</td>
+		<td>3</td>
+		<td>4</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>2</td>
+		<td>4</td>
+		<td>5</td>
+		<td>6</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>3</td>
+		<td>6</td>
+		<td>8</td>
+		<td>10</td>
+	</tr>
+	</table>
+</center>
+
+Notice how all queries can now be processed using 
+
+$$
+\begin{aligned}
+\texttt{arr}[i][j]=&\,\texttt{prefix}[A][B]
+        - \texttt{prefix}[a-1][B] \\
+        &- \texttt{prefix}[A][b-1] + \texttt{prefix}[a-1][b-1]
+\end{aligned}
+$$
+
+For example, if we query $(2, 2)$ to $(3, 4)$ we get $A=3, B=4, a=2, b=2$, and substituting into our equation, we get
+
+$$
+\texttt{prefix}[3][4] - \texttt{prefix}[1][4] - \texttt{prefix}[3][1] + \texttt{prefix}[1][1] = 6 - 1 - 2 - 0 = 3
+$$
+
+#### Forest Queries Solution
```

Comment:
Please separate arr and pfx in implementation. Add a comment saying that they can be the same in practice. However, it helps to explicitly differentiate them in the impl